monitor/monitor.go

```diff
@@ -124,23 +140,100 @@ func (m *Monitor) start() {
 	}
 }
 
-func (m *Monitor) scanKnownJobs() error {
-	jobs, err := m.k8s.BatchV1().Jobs(m.cfg.Namespace).List(m.ctx, v1.ListOptions{
-		LabelSelector: api.DefaultLabel,
-	})
+func (m *Monitor) synchronize(ctx context.Context) error {
+	hasTag, err := labels.NewRequirement(api.TagLabel, selection.In, tagsToLabels(m.cfg.Tags))
+	if err != nil {
+		return fmt.Errorf("failed to create tag label requirement: %w", err)
+	}
+	hasUuid, err := labels.NewRequirement(api.UUIDLabel, selection.Exists, nil)
 	if err != nil {
-		return fmt.Errorf("failed to load jobs: %w", err)
+		return fmt.Errorf("failed to create uuid label requirement: %w", err)
 	}
-	for _, job := range jobs.Items {
-		uuid, found := job.Labels[api.DefaultLabel]
-		if !found {
-			m.logger.Error("job found without label", zap.String("name", job.Name))
-		} else {
-			if job.Status.CompletionTime == nil {
-				m.logger.Debug("adding previously scheduled job", zap.String("uuid", uuid))
-				m.knownBuilds.Add(uuid, struct{}{})
+	selector := labels.NewSelector().Add(*hasTag, *hasUuid).String()
+	lw := &cache.ListWatch{
+		ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
+			options.LabelSelector = selector
+			return m.k8s.BatchV1().Jobs(m.cfg.Namespace).List(ctx, options)
+		},
+		WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
+			options.LabelSelector = selector
+			return m.k8s.BatchV1().Jobs(m.cfg.Namespace).Watch(ctx, options)
+		},
+	}
+	removeIfCompleted := func(obj interface{}) {
```

Comment:
We talked last week about how the buildkite agent has a tendency towards outputting a ton of information for user diagnosis. I wonder if this is a good place to start adding some non-debug logs, especially around job lifecycle.

Reply:
Yeah, I have been pretty much logging everything at debug, but it probably is a good idea to have some non-debug stuff